src/macaron/vsa/vsa.py

```diff
@@ -210,53 +285,39 @@ def get_subject_verification_result(policy_result: dict) -> tuple[str, Verificat
 
     Returns
     -------
-    tuple[str, VerificationResult] | None
-        A pair of PURL and verification result of the only software component that
-        the policy applies to, or ``None`` according to the aforementioned conditions.
+    dict[str, int] | None
+        A dictionary of software components passing the policy, or ``None`` if there is any
+        component failing the policy or if there is no software component in the policy engine result.
+        Each key is a PackageURL of the software component, and each value is the corresponding
+        component id of that component.
     """
     component_violates_policy_facts = policy_result.get("component_violates_policy", [])
     component_satisfies_policy_facts = policy_result.get("component_satisfies_policy", [])
 
+    if len(component_violates_policy_facts) > 0:
+        logger.info("Encountered software component failing the policy. No VSA is generated.")
+        return None
+
     # key: PURL; value: result with the highest component id
```

Comment:
Do the values of this dictionary store the *highest component id number* that has a passing result, or the *result* of the highest component id? 
From my understanding of the implementation, the values are the `highest component id number`, is that correct?

Reply:
I realized I should have updated this comment now that the implementation has changed. Thanks for pointing out.
See my updated comment in 96da6a92e2703fea016413bee29c209ee8655c3c. Let me know if it clarifies things.

Reply:
Yeah, it's clearer too me know. Thanks.